src/internal/ReplaySubject.ts

```diff
@@ -71,18 +73,22 @@ export class ReplaySubject<T> extends Subject<T> {
       subscription = new SubjectSubscription(this, subscriber);
     }
 
-    if (scheduler) {
-      subscriber.add(subscriber = new ObserveOnSubscriber<T>(subscriber, scheduler));
+    if (this.scheduler) {
+      subscriber.add(subscriber = new ObserveOnSubscriber<T>(subscriber, this.scheduler));
     }
 
-    if (_infiniteTimeWindow) {
-      for (let i = 0; i < len && !subscriber.closed; i++) {
-        subscriber.next(<T>_events[i]);
-      }
+    if (this._infiniteTimeWindow) {
+      _events.forEach(event => {
```

Comment:
`for` loop is more efficient, particularly at processing arrays. `forEach` requires a function allocation.